QuickStart/FAQBot/FAQMessageView.swift

```diff
@@ -0,0 +1,184 @@
+    
+    func setupViews() {
+        // +-------------------------------+
+        // | +-----------+---------------+ |
+        // | | imageView | questionLabel | |
+        // | |           +---------------+ |
+        // | |           |  answerLabel  | |
+        // | +-----------+---------------+ |
+        // +-------------------------------+
+        self.addSubview(containerView)
+        
+        containerView.addSubview(containerStackView)
+        
+        [imageView, labelStackView]
+            .forEach { containerStackView.addArrangedSubview($0) }
+        
+        [questionLabel, answerLabel]
+            .forEach { labelStackView.addArrangedSubview($0) }
+    }
+    
+    func setupAutoLayouts() {
+        [containerStackView, labelStackView, containerView, questionLabel, answerLabel, imageView]
+            .forEach { $0.translatesAutoresizingMaskIntoConstraints = false }
+        
+        NSLayoutConstraint.activate([
+            // containerView
+            containerView.topAnchor.constraint(equalTo: self.topAnchor),
+            containerView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            containerView.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            containerView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            containerView.heightAnchor.constraint(equalToConstant: 82),
+            containerView.widthAnchor.constraint(equalToConstant: 244),
+            
+            // container stack view
+            containerStackView.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 12),
+            containerStackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: 12),
+            containerStackView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor, constant: -12),
+            containerStackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -12),
+            
+            // imageView
+            imageView.heightAnchor.constraint(equalToConstant: 56),
+            imageView.widthAnchor.constraint(equalToConstant: 56),
+            
+            // questionLabel
+            questionLabel.heightAnchor.constraint(greaterThanOrEqualToConstant: 20),
+            
+            // answerLabel
+            answerLabel.heightAnchor.constraint(lessThanOrEqualToConstant: 56),
+        ])
+    }
+    
+    func setupActions() {
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(didTap(_:)))
+        self.addGestureRecognizer(tapGesture)
+    }
+    
+    func setupStyles() {
+        questionLabel.text = faqResult.question
+        answerLabel.text = faqResult.answer
+        
+        if let imageURL = URL(string: faqResult.imageURL ?? "") {
+            imageView.load(url: imageURL)
+            imageView.layer.cornerRadius = 10
+            imageView.layer.masksToBounds = true
+        } else {
+            imageView.isHidden = true
+        }
+        
+        
```

Comment:
Can we clean up these double newlines? I see many of them in the code

Reply:
done